core/src/bin/test_auth_flow.rs

```diff
@@ -1,17 +1,21 @@
-use mongo_odbc_core::oidc_auth::*;
+use mongo_odbc_core::oidc_auth::do_auth_flow;
+use mongodb::options::oidc::{CallbackContext, IdpServerInfo};
 
 #[tokio::main]
 async fn main() {
-    let c = CallbackContext {
-        idp_info: Some(IdpServerInfo {
-            issuer: "https://mongodb-dev.okta.com/oauth2/ausqrxbcr53xakaRR357".to_string(),
-            client_id: Some("0oarvap2r7PmNIBsS357".to_string()),
-            // show that we get a refresh_token even when we don't ask for it here
-            request_scopes: Some(vec!["openid".to_string()]),
-        }),
-        refresh_token: None,
-        timeout_seconds: None,
-        version: 1,
-    };
-    println!("{:?}", do_auth_flow(c).await.unwrap());
+    let c = CallbackContext::builder()
+        .idp_info(
+            IdpServerInfo::builder()
+                .issuer("https://mongodb-dev.okta.com/oauth2/ausqrxbcr53xakaRR357".to_string())
```

Comment:
In the JDBC driver I put these repeated URL and client id's in a separate utility file.  You may want to do that as well in case these change in the future it can be changed in one place.